### PR TITLE
Update Infinispan to 10.1.5.Final and update test classes

### PR DESCRIPTION
**Pull request by wburns**
https://github.com/quarkusio/quarkus/issues/7910

### CODE REVIEW

**Comment by gsmet** [comment]
@jmartisk can you check it fixes the metrics issue?

**Comment by wburns** [comment]
It is failing native build saying it can't find the metrics classes. I am not quite sure why yet as we have quarkus smallrye metrics as a dependency.

**Comment by wburns** [comment]
So I fixed the xml issues by tweaking it slightly (we added extra sanity checks that the xml was violating).

However I had to add in a hack to have the appropriate eclipse microprofile apis available, @jmartisk maybe you could help a bit? Infinispan is on 2.1.5 of smallrye metrics where as Quarkus is on 2.4.1. To be honest I am not sure why this wasn't complaining before though either.

**Comment by wburns** [comment]
Disregard my previous comment, it should be good now.

**Review by gsmet — APPROVED** [review]

**Comment by jmartisk** [comment]
Sorry I was already finished for yesterday
> However I had to add in a hack to have the appropriate eclipse microprofile apis available, @jmartisk maybe you could help a bit? Infinispan is on 2.1.5 of smallrye metrics where as Quarkus is on 2.4.1. To be honest I am not sure why this wasn't complaining before though either.

That sounds like you may want to update soon, we should try to make the versions aligned as much as possible. I see Infinispan master is at 2.3.0, which is a bit better. It shouldn't be a critical problem for now because these versions are backward compatible, but for example in June we will have MicroProfile 4.0 with breaking changes. If you need assistance with upgrading metrics in Infinispan, just drop me a message, but I think it shouldn't be a problem

File: extensions/infinispan-embedded/runtime/pom.xml [hunk 1]
```diff
@@ -30,6 +30,11 @@
             <groupId>io.smallrye</groupId>
             <artifactId>smallrye-metrics</artifactId>
         </dependency>
+        <!-- Hack to get it to have the API Infinispan uses available -->
```
Comment (anistor) [inline extensions/infinispan-embedded/runtime/pom.xml]: Could you elaborate why do you call this a hack @wburns ?

Comment (jmartisk) [reply extensions/infinispan-embedded/runtime/pom.xml]: I'd rather ask why aren't we importing the `quarkus-smallrye-metrics` module here instead?

Comment (wburns) [reply extensions/infinispan-embedded/runtime/pom.xml]: > Could you elaborate why do you call this a hack @wburns ?

It isn't anymore, I forgot to remove the comment is all.

> I'd rather ask why aren't we importing the `quarkus-smallrye-metrics` module here instead?

But I can try this instead and see if it still works. Either way I can make a simple PR here in just a bit.

Comment (gsmet) [reply extensions/infinispan-embedded/runtime/pom.xml]: For me, it depends if you need the API or an implementation.

If you just need the API, I wouldn't make the implementation mandatory.

@jmartisk WDYT?

Comment (jmartisk) [reply extensions/infinispan-embedded/runtime/pom.xml]: AFAIK the SmallRye Metrics implementation is mandatory for Infinispan to be able to run (ideally it should be able to run without metrics dependency at all if they are disabled, but that's not the case right now)

Comment (wburns) [reply extensions/infinispan-embedded/runtime/pom.xml]: I created https://github.com/quarkusio/quarkus/pull/7935 to remove the comment. Let me know if I should change the dependency though.

Comment (jmartisk) [reply extensions/infinispan-embedded/runtime/pom.xml]: Other extensions that make use of Metrics declare the `quarkus-smallrye-metrics` dependency in their runtime module, so I'd vote for yes

Comment (gsmet) [reply extensions/infinispan-embedded/runtime/pom.xml]: Ah ah, I missed the `smallrye-metrics` dependency just above. So yes, we need the extension to be used.

Comment (anistor) [reply extensions/infinispan-embedded/runtime/pom.xml]: > right now

Now (10.1.5.final) it should be able to work without Smallrye metrics being present in classpath at runtime. It's truly optional.

**Pull request merged**